src/dataflow/src/source/mod.rs

```diff
@@ -139,13 +139,20 @@ where
     {
         Exchange::new(|x: &Self| x.key.hashed())
     }
-    /// A parallelization contract that hashes by values.
+    /// A parallelization contract that hashes by positions (if available)
+    /// and otherwise falls back to hashing by value.
```

Comment:
Can you explain in the comment why we're doing this / what problems we ran into distributing just by value?

Reply:
sure ill add that